elixir-smie.el

```diff
@@ -364,9 +368,13 @@
         (t (smie-rule-parent elixir-smie-indent-basic))))))
     (`(:before . ";")
      (cond
+      ((and (smie-rule-parent-p "if")
```

Comment:
These rules are getting quite extensive. We need to start adding comments explaining what these are for.